spatial/transform/umeyama.go

```diff
@@ -0,0 +1,146 @@
+// Copyright ©2025 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"errors"
+
+	"gonum.org/v1/gonum/mat"
+	"gonum.org/v1/gonum/stat"
+)
+
+// ErrSVDFailed is returned when a required SVD factorization process fails.
+var ErrSVDFailed = errors.New("transform: SVD factorization failed")
+
+// Umeyama finds the similarity transformation between two sets of points
+// that minimizes the mean squared error between them.
+//
+// The transformation relates two sets of n corresponding points {x_i}
+// and {y_i} as:
+//
+//	y_i ≈ c * R * x_i + t,  i=1,...,n
+//
+// where c is the scale factor, R is the rotation matrix and t is
+// the translation vector.
+//
+// The point sets are represented as two n×m matrices X and Y, where
+// m is the number of dimensions and x_i and y_i are stored in the i-th
+// row of X and Y, respectively. Typically, m is equal to 2 or 3.
+// If the dimensions of X and Y are not equal, Umeyama will panic.
+//
+// Umeyama returns the scale factor c, the rotation matrix r and the translation
```

Comment:
Just a nitpick, I'd refer to the mathematical entity R here:
```suggestion
// Umeyama returns the scale factor c, the rotation matrix R and the translation
```